app/models/user.rb

```diff
@@ -14,11 +14,13 @@ class User < ApplicationRecord
     validates :address, presence: true
     validates :city, presence: true, format: {with: /[A-Z]\D+/, message: "is invalid, please use Uppercase in first letter"}
     validates :postalcode, presence: true, format: {with: /\b\d{5}\b/, message: "is invalid, please enter a correct Postal Code"}
+
     validate  :is_adult?
 
     enum :doc_kind, {dni: 0, driver_license: 1, passport: 2}
+    enum :role, {admin: 00, customer: 01, premium_customer: 10, blocked: 0, banned: 1}
+    after_initialize :set_default_role, :if => :new_record?
```

Comment:
great trick! 👏🏼 this is right. there is an alternative way to do this directly with the enum. are you able to tell me which one is?

Reply:
with the _default option!
enum :role, {admin: 00, customer: 01, premium_customer: 10, blocked: 0, banned: 1}, _default: :customer

Had to look it up because i didnt knew it ,but seems very handy and readable